### PR TITLE
Add pulse functionality to light bulbs.

### PR DESCRIPTION
**Pull request by chasdevs**
When a meeting just ended and there is one coming up, pulse red. When a meeting just started, pulse green.

### CODE REVIEW

File: src/test/trafficLights/unit/MeetingTrafficLights.test.ts [hunk 1]
```diff
@@ -19,43 +21,74 @@ describe('MeetingTrafficLights Unit Test', () => {
...
+        afterEach(td.reset)
+        beforeEach(() => {
+            now = dayjs()
+            bulb = Factory.dummyBulb('My Meeting Room')
+            room = Factory.dummyRoom('My Meeting Room')
+            td.when(mockNetwork.scanForBulbs()).thenResolve([bulb]);
+            td.when(mockCalendarService.rooms()).thenResolve([room]);
+        })
+
+        test('meeting in progress', async () => {
```
Comment (chasdevs) [inline src/test/trafficLights/unit/MeetingTrafficLights.test.ts]: Refactored into individual tests.

File: src/test/trafficLights/unit/MeetingTrafficLights.test.ts [hunk 2]
```diff
@@ -21,7 +21,46 @@ describe('MeetingTrafficLights Unit Test', () => {
...
+        // Configure mocks
+        const bulbs = Factory.dummyBulbs([{label: 'meeting exists'}, {label: 'ending soon'}, {label: 'GTfo'}, {label: 'no meetings'}, {label: 'just started'}])
+        const [bulbWithMeeting, bulbEndingSoon, bulbEnded, bulbNoMeeting, bulbJustStarted] = bulbs;
+        const rooms = Factory.dummyRooms([{name: 'Meeting exists!'}, {name: 'Ending Soon...'}, {name: 'GTFO'}, {name: 'no meetings here'}, {name: 'just started'}])
+        const [roomWithMeeting, roomEndingSoon, roomEnded, roomNoMeeting, roomJustStarted] = rooms;
+
+        td.when(mockNetwork.scanForBulbs()).thenResolve(bulbs);
+        td.when(mockCalendarService.rooms()).thenResolve(rooms);
+        td.when(mockCalendarService.getCurrentMeetings(roomWithMeeting)).thenResolve({
+            currentMeeting: Factory.dummyMeeting({location: roomWithMeeting, start: now.subtract(6, 'minute'), end: now.add(5, 'minute').add(1, 'second')})
```
Comment (Zugwalt) [inline src/test/trafficLights/unit/MeetingTrafficLights.test.ts]: why not subtract 59 seconds?

File: src/test/trafficLights/unit/MeetingTrafficLights.test.ts [hunk 3]
```diff
@@ -21,7 +21,46 @@ describe('MeetingTrafficLights Unit Test', () => {
...
+
+        // Call test method
+        await app.syncBulbs()
+
+        // Verify results
+        td.verify(bulbWithMeeting.setColor(Color.GREEN_SOFT))
+        td.verify(bulbEndingSoon.setColor(Color.ORANGE))
+        td.verify(bulbEnded.setColor(Color.RED))
+        td.verify(bulbNoMeeting.powerOn(false))
+        td.verify(bulbJustStarted.pulse(Color.GREEN))
```
Comment (Zugwalt) [inline src/test/trafficLights/unit/MeetingTrafficLights.test.ts]: Nice mocking and stuff--we got a real test now!

File: src/app/bulbNetworks/Bulb.ts [hunk 4]
```diff
@@ -4,5 +4,6 @@ export default interface Bulb {
     readonly id: string;
     readonly label: string;
     setColor: (color: Color) => Promise<void>
+    pulse: (color: Color) => Promise<void>
```
Comment (Zugwalt) [inline src/app/bulbNetworks/Bulb.ts]: 🚨

File: src/app/bulbNetworks/magicHome/MagicHomeBulb.ts [hunk 5]
```diff
@@ -24,4 +23,8 @@ export default class MagicHomeBulb implements Bulb {
         return this._client.setMagicHomeBulbPower(this._ip, on)
     }
 
+    async pulse(color: Color): Promise<void> {
+        throw new Error('pulse not implemented for MagicHome')
```
Comment (Zugwalt) [inline src/app/bulbNetworks/magicHome/MagicHomeBulb.ts]: 📉

File: src/app/trafficLights/MeetingTrafficLights.ts [hunk 6]
```diff
@@ -85,20 +85,28 @@ export default class MeetingTrafficLights {
...
+         *                                            |
+         *                                 (pulse-red if meeting next)
          */
-        if (previousMeeting && now.diff(previousMeeting.end, 'minute') < this._endIntervalMin
-            || currentMeeting && currentMeeting.end.diff(now, 'minute') == 0) {
-            logger.debug('Meeting over! Setting bulb to Red.', {room: room.name, bulb: bulb.label})
+        if (currentMeeting && previousMeeting && now.diff(previousMeeting.end, 'm') < this._endIntervalMin
+            || currentMeeting && now.diff(currentMeeting.end, 'm') == 0 && nextMeeting && nextMeeting.start.diff(now, 'm') == 0) {
+            logger.debug('Meeting over, time to get out! Pulsing Red.', {room: room.name, bulb: bulb.label})
+            await bulb.pulse(Color.RED)
```
Comment (Zugwalt) [inline src/app/trafficLights/MeetingTrafficLights.ts]: Nice!

**Comment by chasdevs** [comment]
Good idea. I'll add config options.

**Pull request merged**